OneWelcomeExampleApp/ViewControllers/QR-code/QRCodeViewController.swift

```diff
@@ -51,21 +51,68 @@ class QRCodeViewController: UIViewController {
         super.viewDidLoad()
 
         view.backgroundColor = UIColor.black
-        captureSession = AVCaptureSession()
+        
+        setupCaptureSession()
+        startSession()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
 
+        if captureSession?.isRunning == false {
```

Comment:
actually I don't like that ViewControllers holds everything (Massive View Controller)...
captusSession/preview Layer and its logic should be done in separate layers (interactor/presenter).

This would require a bit more work, but you started with the refactor already, you can continue it in proper way.

Reply:
if you don't know how to proceed, please create a separate story for the refactor and I will take it.

But I hope that after you catch the flow in oad-907, this should be doable for you

Reply:
Actually the aim was not to refactor but to fix main thread issue. So lets merge this one and create a separate story for refactor.

Reply:
TLDR please resolve the comment and approve if you don't have any other remarks.

Reply:
ok